services/web/server/src/async_sio.py

```diff
@@ -121,27 +121,21 @@ def stop_dynamic_service(sid, data):
 
 @SIO.on('listObjects')
 async def list_S3_objects(sid, data):
-    _LOGGER.debug("client %s requests S3 objects in %s", sid, data)
+    _LOGGER.debug("client %s requests objects in storage. Extra argument %s", sid, data)
     _config = s3_config()
 
     s3_client = S3Client(endpoint=_config.endpoint,
         access_key=_config.access_key, secret_key=_config.secret_key)
 
     objects = s3_client.list_objects_v2(_config.bucket_name)
+    dataOut = []
     for obj in objects:
-        dataOut = {}
-        dataOut['name'] = obj.object_name
-        dataOut['lastModified'] = json.dumps(obj.last_modified, indent=4, sort_keys=True, default=str)
-        dataOut['size'] = obj.size
-        await SIO.emit('listObjectsPub', data=dataOut, room=sid)
-
-    #objects = s3_client.list_objects_v2(data)
-    #for obj in objects:
-    #    dataOut = {}
-    #    dataOut['name'] = obj.object_name
-    #    dataOut['lastModified'] = json.dumps(obj.last_modified, indent=4, sort_keys=True, default=str)
-    #    dataOut['size'] = obj.size
-    #    await SIO.emit('listObjectsUser', data=dataOut, room=sid)
+        objInfo = {}
+        objInfo['path'] = obj.bucket_name + '/' + obj.object_name
+        objInfo['lastModified'] = json.dumps(obj.last_modified, indent=4, sort_keys=True, default=str)
```

Comment:
Is there a real need to format the json with indent and sorting keys? That is normally done to make it human-readable while here we intend to transmit it, right?

Reply:
@odeimaiz and what about this?

Reply:
@pcrespov I will be happy to listen to other solutions

Reply:
@odeimaiz check for instance this https://gist.github.com/drmalex07/5149635e6ab807c8b21e

Reply:
@pcrespov no worries, fixed